server/controllers/routes/post/put.js

```diff
@@ -0,0 +1,65 @@
+const { join } = require('path');
+
+const { eventSchema, publicServiceSchema } = require('../../utils/postSchema');
+const {
+  updateEventQuery,
+  updateTopicQuery,
+  updatePublicServiceQuery,
+  updateSecondaryTagQuery,
+} = require('../../../database/queries/updatePost');
+
+const updateEvent = (req, res, next) => {
+  const { postId: eventId } = req.params;
+  const { eventTopic } = req.body;
+  const { image } = req.files;
+  let imageName = '';
+  if (image) {
+    imageName = Date.now() + image.name;
+    image.mv(join(__dirname, '..', '..', '..', 'uploads', imageName), (err) => {
+      if (err) next(err);
+    });
+  }
+
+  eventSchema.isValid(req.body).then((valid) => {
+    if (!valid) res.status(400).send({ error: 'Bad Request', statusCode: 400 });
+    else {
+      updateEventQuery(eventId, req.body, imageName)
+        .then(() => eventTopic.forEach(topic => updateTopicQuery(eventId, topic)))
+        .then(() => res.send({ data: 'Updated event successfully', statusCode: 200 }))
+        .catch(next);
+    }
+  });
+};
+
+const updatePublicService = (req, res, next) => {
+  const { postId: publicServiceId } = req.params;
+  const { secondaryTag } = req.body;
+  let imageName = '';
+  if (req.files) {
+    const { image } = req.files;
+    imageName = Date.now() + image.name;
+    image.mv(join(__dirname, '..', '..', '..', 'uploads', imageName), (err) => {
```

Comment:
image.mv is async function, so you will send response with 200 without ensure either the image saved or not 
```suggestion
    image.mv(join(__dirname, '..', '..', '..', 'uploads', imageName), (err) => {
```